### PR TITLE
spline #761 ArangoDB: Add WITH keywords and replace UDFs with Foxx services

### PR DESCRIPTION
**Pull request by wajda**
This PR adds `WITH` keyword to all AQL queries

### CODE REVIEW

**Review by cerveada — APPROVED** [review]
LGTM

**Comment by wajda** [comment]
Since more issues were found I am withdrawing the PR for now until all related issues are fixed.

File: persistence/src/main/resources/AQLFunctions/README.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Spline ArangoDB AQL functions are defined in JS files in the following pattern:
```
Comment (cerveada) [inline persistence/src/main/resources/AQLFunctions/README.txt:1]: Maybe we should mention the AQL limitations here

**Review by cerveada — APPROVED** [review]
LGTM

File: persistence/src/main/resources/AQLFunctions/README.txt (hunk 1 above)
Comment (wajda) [reply persistence/src/main/resources/AQLFunctions/README.txt:1]: Let's assume we're the last ones who don't read manuals :)

**Pull request merged**